### ISSUE TITLE
Look into integrating with Ember Simple Auth 

### ISSUE THREAD

**Issue by jwngr** [firebase/emberfire#111]
http://ember-simple-auth.simplabs.com/

**Comment by tstirrat** [comment]
Please look to the [torii-fire](https://github.com/MattMSumner/torii-fire) library for integrating auth. It should be usable via the ember-simple-auth torii adapter.

**Comment by jamesdixon** [comment]
If anyone's interested, I just released a custom Firebase authenticator for Ember Simple Auth. You can find it here: https://github.com/jamesdixon/ember-cli-simple-auth-firebase

At this time, it only supports Firebase's email + password authentication.

**Comment by ksnyde** [comment]
where has this gotten to? I'm trying to use ember-simple-auth 1.0 (fully rewritten from prior versions) and want to plugin firebase but can't quite get my head around it.

**Comment by jamesdixon** [comment]
@ksnyde yeah, i hope it works, but can't make any guarantees. Unfortunately, I'm not using Firebase any longer, so I have no plans to update it. That said, more than willing to accept a PR if it needs/you want to update it.

**Comment by cs3b** [comment]
@ksnyde did you get it up and running? 

So far I was able to configure 

ember-simple-auth with torii ( I've generated authenticator ) next I wrote my component 

````js
// layout/authentication-widget/component.js
import Ember from 'ember';

export default Ember.Component.extend({
  session: Ember.inject.service(),

  actions: {
    signIn: function (provider = "google") {
      this.get("session").authenticate('authenticator:torii', 'firebase', {provider: provider});
    },
    signOut: function () {
      this.get("session").invalidate();
    }
  }
});
````

And the popup invocation works good - I can make authentication with google, and when popup is closing I'm getting:

````
onTransportReturned@http://localhost:4200/assets/vendor.js:76679:17
onMessage@http://localhost:4200/assets/vendor.js:76065:15
EventListener.handleEvent*ns.wrapper/fb.login.transports.util.addListener@http://localhost:4200/assets/vendor.js:75796:7
ns.wrapper/fb.login.transports.Popup.prototype.open@http://localhost:4200/assets/vendor.js:76074:3
ns.wrapper/fb.login.AuthenticationManager.prototype.requestWithTransports_@http://localhost:4200/assets/vendor.js:76674:3
ns.wrapper/fb.login.AuthenticationManager.prototype.authWithTransports_@http://localhost:4200/assets/vendor.js:76642:3
ns.wrapper/fb.login.AuthenticationManager.prototype.authWithPopup@http://localhost:4200/assets/vendor.js:76547:3
ns.wrapper/Firebase<.authWithOAuthPopup@http://localhost:4200/assets/vendor.js:80328:3
_toPromise/<@http://localhost:4200/assets/vendor.js:101747:9
initializePromise@http://localhost:4200/assets/vendor.js:63666:7
Promise@http://localhost:4200/assets/vendor.js:65516:7
_toPromise@http://localhost:4200/assets/vendor.js:101724:14
open@http://localhost:4200/assets/vendor.js:101707:18
proxyToProvider/</<@http://localhost:4200/assets/vendor.js:118475:17
initializePromise@http://localhost:4200/assets/vendor.js:63666:7
Promise@http://localhost:4200/assets/vendor.js:65516:7
proxyToProvider/<@http://localhost:4200/assets/vendor.js:118474:14
authenticate@http://localhost:4200/assets/vendor.js:98622:14
authenticate@http://localhost:4200/assets/vendor.js:98978:14
authenticate@http://localhost:4200/assets/vendor.js:99711:14
signIn@http://localhost:4200/assets/merit-money.js:118:9
Component<.send@http://localhost:4200/assets/vendor.js:51658:28
````

Not sure what is the problem - hard for me to debug.

**Comment by ksnyde** [comment]
I decided I didn't need simple-auth and just used torii to firebase. Been using it for several months now and it works well.

**Comment by cs3b** [comment]
@ksnyde I figure out what was the problem - to make long story short - issue was with google credential, one letter was missing (my bad). Still debugging was hard for me ( first time I was looking into `ember-simple-auth` and `torii` source code). My bad was that I didn't capture reject on promise that `authenticate` return - it would be more easier to spot the problem.


In the end it looks very good - all that I did before was done correctly: authenticator, torii-adapter, component (and it works as I expect). I picked ember-simple-auth as I know that in the end I might not be using firebase authentication (before in one of the project I've use torii session service, just like you, and it works fine).

If ever would need any assist, just let me know - I might write a tutorial how to setup this one, as I didn't found anything in the internet.
